agent/engine/docker_task_engine.go

```diff
@@ -657,6 +663,35 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 	return engine.serialPull(task, container)
 }
 
+// useLocalImageCache returns true if local image cache should be used, or return false to continue
+// pulling image, by inspecting the agent pull behavior variable defined in config. The caller has
+// to make sure the container passed in is not an internal container.
+func (engine *DockerTaskEngine) useLocalImageCache(agentPullBehavior config.AgentPullBehaviorType,
+	container *api.Container,
+	taskArn string) bool {
+	// If the agent pull behavior is never, we don't try to pull the image,
+	// try to use local image cache instead.
+	if agentPullBehavior == config.NeverAgentPullBehavior {
+		seelog.Infof("Task engine [%s]: use cached image directly for container %s",
+			taskArn, container.Name)
+		// It's not an internal container, so it's safe to add container reference.
+		engine.updateContainerReference(false, container, taskArn)
```

Comment:
@richardpen and I discussed about the problem we may face about image cleaning. If we use `never` or `once` agent behavior setup, the image cleanup process could remove the image, and I think for `never` and `once` we should keep the image cache, right?

We think what we can do: 
1. Let customer know that there may be conflicts between image cleaning and agent pull behavior setup, and suggest them when using `never` or `once`, set `NumImagesToDeletePerCycle` to be 0 (or some other image cleanup settings that can prevent image from being removed).

2. Disable image cleanup internally when using `never` or `once`, add some documentation on that so that user knows.

@sharanyad @richardpen @aaithal what do you think about this?